### PR TITLE
task1

### PR DESCRIPTION
**Pull request by EvgeniyaSirenko**

### CODE REVIEW

File: src/main/java/com/epam/preprod/sirenko/entity/Clothing.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.Objects;
+
+/**
+ * Clothing entity.
+ *
+ * @author E.Sirenko
+ **/
+public class Clothing extends Products {
+	private String size;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Clothing.java]: I think better approach for such field is to create enum with different sizes than use string type

File: src/main/java/com/epam/preprod/sirenko/entity/Clothing.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	private String season;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Clothing.java]: Same

File: src/main/java/com/epam/preprod/sirenko/entity/Food.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+import java.util.Objects;
+
+/**
+ * Food entity.
+ *
+ * @author E.Sirenko
+ **/
+public class Food extends Products {
+	private int amount;
+	private String petType;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Food.java]: Do we need 'petType' field of Food? What does it mean? Maybe we need another class for this case? For example something like PetFood that extend Food

File: src/main/java/com/epam/preprod/sirenko/entity/Food.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+import java.util.Objects;
+
+/**
+ * Food entity.
+ *
+ * @author E.Sirenko
+ **/
+public class Food extends Products {
+	private int amount;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Food.java]: Maybe we should rename this field to quantity or weight to better describe behavior?

File: src/main/java/com/epam/preprod/sirenko/entity/Products.java [hunk 5]
```diff
@@ -1,30 +1,24 @@
...
+package com.epam.preprod.sirenko.entity;
 
 import java.util.Objects;
 
+/**
+ * Products entity.
+ *
+ * @author E.Sirenko
+**/
 public abstract class Products {
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Products.java]:
```suggestion
public abstract class Product {
```

File: src/main/java/com/epam/preprod/sirenko/entity/Products.java [hunk 6]
```diff
@@ -1,30 +1,24 @@
...
-	
-	int id;
-	String name;
-	int price;
+	private String name;
+	private int price;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Products.java]: Maybe we should use BigDecimal as a type for price?

File: src/main/java/com/epam/preprod/sirenko/entity/Products.java [hunk 7]
```diff
@@ -1,30 +1,24 @@
...
 	
 	public Products() {
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Products.java:14]: Do we need constructors for abstract class?

File: src/main/java/com/epam/preprod/sirenko/entity/Products.java [hunk 8]
```diff
@@ -1,30 +1,24 @@
...
 	}
 	
-	public Products(int id, String name, int price) {
-		this.id = id;
+	public Products(String name, int price) {
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Products.java]: Same

File: src/main/java/com/epam/preprod/sirenko/entity/DryFood.java [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+
+import java.util.Objects;
+
+/**
+ * DryFood entity.
+ *
+ * @author E.Sirenko
+ **/
+public class DryFood extends Food {
+	private int petAge;
```
Comment (VitaliiZahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/DryFood.java]: What do fields petAge and petWeight mean for DryFood?

File: src/main/java/com/epam/preprod/sirenko/entity/Clothing.java (hunk 1 above)
Comment (EvgeniyaSirenko) [reply src/main/java/com/epam/preprod/sirenko/entity/Clothing.java]: done

File: src/main/java/com/epam/preprod/sirenko/entity/Clothing.java (hunk 2 above)
Comment (EvgeniyaSirenko) [reply src/main/java/com/epam/preprod/sirenko/entity/Clothing.java]: done

File: src/main/java/com/epam/preprod/sirenko/entity/Food.java (hunk 3 above)
Comment (EvgeniyaSirenko) [reply src/main/java/com/epam/preprod/sirenko/entity/Food.java]: refactored it to petGroup enum

File: src/main/java/com/epam/preprod/sirenko/entity/Food.java (hunk 4 above)
Comment (EvgeniyaSirenko) [reply src/main/java/com/epam/preprod/sirenko/entity/Food.java]: renamed to weight

File: src/main/java/com/epam/preprod/sirenko/entity/Clothing.java [hunk 10]
```diff
@@ -0,0 +1,62 @@
...
+import java.util.Objects;
+
+/**
+ * Clothing entity.
+ *
+ * @author E.Sirenko
+ *
+ **/
+public class Clothing extends Product {
+	private int sizeId;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Clothing.java]:
```suggestion
	private Season season;
```

File: src/main/java/com/epam/preprod/sirenko/entity/DryFood.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+import java.util.Objects;
+
+/**
+ * DryFood entity.
+ *
+ * @author E.Sirenko
+ **/
+public class DryFood extends Food {
+	private String brandName;
+	private int petGroupId;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/DryFood.java]:
```suggestion
	private PetGroup pet;
```

File: src/main/java/com/epam/preprod/sirenko/entity/PetGroup.java [hunk 12]
```diff
@@ -0,0 +1,20 @@
...
+/**
+ * PetType entity.
+ *
+ * @author E.Sirenko
+ *
+ **/
+public enum PetGroup {
+	CAT, DOG, SMALL_PET;
+	
+	public static PetGroup getPetGroup(DryFood dryFood) {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/PetGroup.java]: We dont need this method, you can delete it here and same for other enums

File: src/main/java/com/epam/preprod/sirenko/entity/PetGroup.java [hunk 13]
```diff
@@ -0,0 +1,20 @@
+package com.epam.preprod.sirenko.entity;
+
+/**
+ * PetType entity.
+ *
+ * @author E.Sirenko
+ *
+ **/
+public enum PetGroup {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/PetGroup.java]: Create separete directory for enums (entity.enum for example)

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 14]
```diff
@@ -0,0 +1,244 @@
...
+import java.util.Spliterators;
+import java.util.function.UnaryOperator;
+
+/**
+ * Resizable array-implementation of the List interface
+ *
+ * @author E.Sirenko
+ *
+ **/
+public class Container implements List {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Move Container to separate directory

File: src/main/java/com/epam/preprod/sirenko/entity/DryFood.java [hunk 15]
```diff
@@ -0,0 +1,61 @@
...
+	
+	public DryFood() {
+	}
+	
+	public DryFood(String name, BigDecimal price, int foodWeight, String dryFoodBrandName, int dryFoodPetGroupId) {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/DryFood.java]: Also fix constructors for enum fields

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java (hunk 14 above)
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Container should hold only inheritors of Product

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 16]
```diff
@@ -0,0 +1,244 @@
...
+	private int size = 10;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: We dont interact with this field in both constructors

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 17]
```diff
@@ -0,0 +1,244 @@
...
+		size++;
+		return true;
+	}
+	
+	/**
+	 *
+	 * Inserts the specified element at the specified position
+	 **/
+	@Override
+	public void add(int index, Object o) {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Please use meaningful names of variables for readability then just o

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 18]
```diff
@@ -0,0 +1,244 @@
...
+		return array.length;
+	}
+	
+	/**
+	 *
+	 * Removes all the elements
+	 **/
+	@Override
+	public void clear() {
+		for (int i = 0; i < size; i++)
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: We can just recreate array

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 19]
```diff
@@ -0,0 +1,244 @@
...
+	}
+	
+	/**
+	 *
+	 * Returns an array containing all of the elements in this container
+	 * in proper sequence (from first to last element).
+	 **/
+	@Override
+	public Object[] toArray() {
+		return array.clone();
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: you can use Arrays.copyOf

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+	}
+	
+	/**
+	 *
+	 * Returns an array containing all of the elements in this list
+	 **/
+	@Override
+	public Object[] toArray(Object[] a) {
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Please dont use such naming like 'a'

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+	
+	@Override
+	public Iterator iterator() {
+		return null;
+	}
+	
+	@Override
+	public ListIterator listIterator() {
+		//no realisation needed
+		return null;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]:
```suggestion
		
`throw new UnsupportedOperationException();`
```

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+	}
+	
+	@Override
+	public ListIterator listIterator(int index) {
+		//no realisation needed
+		return null;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Same

File: src/main/java/com/epam/preprod/sirenko/entity/Container.java [hunk 23]
```diff
@@ -0,0 +1,244 @@
...
+	}
+	
+	@Override
+	public List subList(int fromIndex, int toIndex) {
+		//no realisation needed
+		return null;
```
Comment (v-zahoruiko) [inline src/main/java/com/epam/preprod/sirenko/entity/Container.java]: Same

**Review by v-zahoruiko — APPROVED** [review]

**Pull request merged**